### PR TITLE
Versions changes

### PR DESCRIPTION
**Pull request by nirarg**
**What this PR does / why we need it**:
1. Use capi version v1beta1 (changed from v1alpha4)
2. Use Go version 1.17 (changed from 1.16)

### CODE REVIEW

File: config/default/manager_image_patch.yaml
Comment (rmohr) [inline config/default/manager_image_patch.yaml:11]: This change seems to be not intentional.

File: go.mod
Comment (rmohr) [inline go.mod:74]: There are a lot of replaces. What exactly is the reason for this? Is it the v1beta1 api from capi?

**Review by rmohr — CHANGES_REQUESTED** [review]
Look pretty good overall. A few comments inline.

File: go.mod
Comment (nirarg) [reply go.mod:74]: The section you pointed on is "require" section
This is auto created after the update to go 1.17
There are 2 `require` sections which are auto generated 
There is one `replace` section which include only 3 lines

File: config/default/manager_image_patch.yaml
Comment (nirarg) [reply config/default/manager_image_patch.yaml]: You right, removed this change

File: go.mod
Comment (rmohr) [reply go.mod:74]: > The section you pointed on is "require" section

Indeed :+1:

**Review by davidvossel — APPROVED** [review]
/lgtm

I don't have lgtm permissions yet, but take that as an offer of my support :D 

The code looks good to me and func tests passed for me locally. We'll need an OWNER to issue the `/ok-to-test` command in order for CI to execute the e2e tests

**Comment by rmohr** [comment]
@nirarg  I need to fix something in the integration test. I have to bump it to 1.17 too: 

```
../../../../pkg/mod/k8s.io/client-go@v0.23.0-alpha.4/plugin/pkg/client/auth/exec/metrics.go:21:2: package io/fs is not in GOROOT (/root/.gimme/versions/go1.15.8.linux.amd64/src/io/fs)
```

**Comment by nirarg** [comment]
> @nirarg I need to fix something in the integration test. I have to bump it to 1.17 too:

I rebased the PR on this fix and changed Golang version in the Dockerfile

**Comment by nirarg** [comment]
/hold
Merge decreasing to v1alpha1 version in different PR:
https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/54

**Comment by nirarg** [comment]
/hold cancel
@cchengleo I rebased this PR after #54 was merged
Can you please test this too?

**Review by cchengleo — APPROVED** [review]
I also verified this PR in our CI and passed. Thanks for the great job!

**Pull request merged**
